exploration/semantic-skeletons.md

```diff
@@ -0,0 +1,180 @@
+  However, this means that the translator has to modify the string to get localized output.
+  For example, here are some picture strings with their output vs. the skeleton `MMMddyyyy`:
+```
+en-US
+MMM dd, yyyy=Apr 21, 2025
+dd MMM, yyyy=21 Apr, 2025
+MM/dd/yyyy=04/21/2025
+dd-MM-yyyy=21-04-2025
+Apr 21, 2025
+---
+fr-FR
+MMM dd, yyyy=avr. 21, 2025
+dd MMM, yyyy=21 avr., 2025
+MM/dd/yyyy=04/21/2025
+dd-MM-yyyy=21-04-2025
+21 avr. 2025
+---
+ja-JP
+MMM dd, yyyy=4月 21, 2025
+dd MMM, yyyy=21 4月, 2025
+MM/dd/yyyy=04/21/2025
+dd-MM-yyyy=21-04-2025
+2025年4月21日
+```
+
+
+## Background
+
+_What context is helpful to understand this proposal?_
+
+Links:
+- [Semantic Skeletons Specification](https://unicode.org/reports/tr35/tr35-dates.html#Semantic_Skeletons)
+- [ICU4X Field Set Enum \(strongly typed\)](https://unicode-org.github.io/icu4x/rustdoc/icu/datetime/fieldsets/enums/enum.CompositeFieldSet.html)
+- [ICU4X Field Set Builder \(more JS-like\)](https://unicode-org.github.io/icu4x/rustdoc/icu/datetime/fieldsets/builder/struct.FieldSetBuilder.html)
+
+
+Semantic skeletons are not the first attempt to provide this functionality.
+Previous skeleton mechanisms used 
+collections of field options (as in `Intl.DateTimeFormat`)
+or a microsyntax (as in ICU4J).
+
+The `Intl.DateTimeFormat` skeletons consist of "option bags"
+such as `{ year: "numeric", month: "short", day: "numeric" }`
+in which the user specifies the field and its width.
+Only fields appearing in the options appear in the formatted date/time value.
+
+The ICU microsyntax uses strings supplied by the developers.
+These strings specify the fields and field lengths that should appear in the formatted value.
+See [here](https://unicode-org.github.io/icu/userguide/format_parse/datetime/#date-field-symbol-table)
+The system then uses the string to perform date/time pattern generation,
+arranging the specified fields in the correct order,
+selecting locale-appropriate separators,
+and producing a "picture string" that can be consumed by date/time formatters
+such as `java.text.SimpleDateFormat`.
+
+## Use-Cases
+
+_What use-cases do we see? Ideally, quote concrete examples._
+
+As a developer, I want to format date, time, or date/time values to show specific fields
+with specific appearance without having to learn a complex microsyntax
+or modify my code or formatting directions for each locale.
+
+As a translator, I want to understand what output a given date/time placeholder will produce 
+in my language.
+
+As a translator, I don't want to have to "translate" or modify a date/time placeholder to suit
+my language's needs.
+I should trust that the placeholder will produce appropriate results for my language.
+
+## Requirements
+
+_What properties does the solution have to manifest to enable the use-cases above?_
+
+1. It should be possible to format common incremental time types
+   (e.g. milliseconds since epoch times)
```

Comment:
This is an odd one to have as the very first requirement. I would challenge that it's _at all_ common to need to format epoch milliseconds in a localizable manner.

Reply:
"Incremental time types" are things like Javascript `Date`, Java `java.util.Date`, `time_t`, etc.

I would find it extremely odd if such types could not be formatted. I should change the parenthetical to be clearer. Formatting random `long` values is not what I meant.

Reply:
Ah; this is meant to be about input, rather than output. The phrasing confused me.

Reply:
I find this to be a requirement looking for a problem. _Of course_ if a language has a data type that represents a date, it should be able to be formatted with a MessageFormat date formatter.

Reply:
@sffc It might be possible to interpret this as only supporting Temporal values. It does no harm to spell out the need to format the various kinds of "classical" timekeeping.